include/libopencm3/stm32/fdcan.h

```diff
@@ -0,0 +1,1420 @@
+#define FDCAN_ILS_RF1WL BIT5
+/* Rx FIFO 1 new message interrupt line */
+#define FDCAN_ILS_RF1NL BIT4
+/* Rx FIFO 0 message lost interrupt line */
+#define FDCAN_ILS_RF0LL BIT3
+/* Rx FIFO 0 full interrupt line */
+#define FDCAN_ILS_RF0FL BIT2
+/* Rx FIFO 0 watermark reached interrupt line */
+#define FDCAN_ILS_RF0WL BIT1
+/* Rx FIFO 0 new message interrupt line */
+#define FDCAN_ILS_RF0NL BIT0
+
+/* --- FDCAN_ILE: FDCAN interrupt line enable register ---------------------- */
+/* Enable interrupt Line 1 */
+#define FDCAN_ILE_EINT1 BIT1
+/* Enable interrupt Line 0 */
+#define FDCAN_ILE_EINT0 BIT0
+
+/* --- FDCAN_GFC: FDCAN global filter configuration register ---------------- */
+/* Accept non-matching frames standard */
+enum fdcan_gfc_anfs {
+	FDCAN_GFC_ANFS_ACCEPT_FIFO0	= 0x0,
+	FDCAN_GFC_ANFS_ACCEPT_FIFO1	= 0x1,
+	FDCAN_GFC_ANFS_REJECT		= 0x2,
+	FDCAN_GFC_ANFS_MASK			= 0x3,
+};
+#define FDCAN_GFC_ANFS_SHIFT 4
+/* Accept non-matching frames extended */
+enum fdcan_gfc_anfe {
+	FDCAN_GFC_ANFE_ACCEPT_FIFO0	= 0x0,
+	FDCAN_GFC_ANFE_ACCEPT_FIFO1	= 0x1,
+	FDCAN_GFC_ANFE_REJECT		= 0x2,
+	FDCAN_GFC_ANFE_MASK			= 0x3,
+};
+#define FDCAN_GFC_ANFE_SHIFT 2
+/* Reject remote frames standard */
+#define FDCAN_GFC_RRFS BIT1
+/* Reject remote frames extended */
+#define FDCAN_GFC_RRFE BIT0
+
+/* --- FDCAN_SIDFC: FDCAN standard ID filter configuration register --------- */
+/* List size standard */
+#define FDCAN_SIDFC_LSS_MASK 0xFF
+#define FDCAN_SIDFC_LSS_SHIFT 16
+/* Filter list standard start address */
+#define FDCAN_SIDFC_FLSSA_MASK 0x3FFF
+#define FDCAN_SIDFC_FLSSA_SHIFT 2
+
+/* --- FDCAN_XIDFC: FDCAN extended ID filter configuration register --------- */
+/* List size extended */
+#define FDCAN_XIDFC_LSE_MASK 0xFF
+#define FDCAN_XIDFC_LSE_SHIFT 16
+/* Filter list standard start address */
+#define FDCAN_XIDFC_FLESA_MASK 0x3FFF
+#define FDCAN_XIDFC_FLESA_SHIFT 2
+
+/* --- FDCAN_XIDAM: FDCAN extended ID and mask register --------------------- */
+/* Extended ID Mask */
+#define FDCAN_XIDAM_EIDM_MASK 0x1FFFFFFF
+#define FDCAN_XIDAM_EIDM_SHIFT 0
+
+/* --- FDCAN_HPMS: FDCAN high priority message status register -------------- */
+/* Filter list */
+#define FDCAN_HPMS_FLST BIT15
+/* Filter index */
+#define FDCAN_HPMS_FIDX_MASK 0x7F
+#define FDCAN_HPMS_FIDX_SHIFT 8
+/* Message Storage Indicator */
+enum fdcan_hpms_msi {
+	FDCAN_HPMS_MSI_NO_FIFO	= 0x0,
+	FDCAN_HPMS_MSI_OVERRUN	= 0x1,
+	FDCAN_HPMS_MSI_FIFO0	= 0x2,
+	FDCAN_HPMS_MSI_FIFO1	= 0x3,
+	FDCAN_HPMS_MSI_MASK		= 0x3,
+};
+#define FDCAN_HPMS_MSI_SHIFT 6
+/* Buffer index */
+#define FDCAN_HPMS_BIDX_MASK 0x3F
+#define FDCAN_HPMS_BIDX_SHIFT 0
+
+/* --- FDCAN_NDAT1: FDCAN new data 1 register ------------------------------- */
+/* New data[31:0] */
+#define FDCAN_NDAT1_ND_MASK 0xFFFFFFFF
+#define FDCAN_NDAT1_ND_SHIFT 0
+
+/* --- FDCAN_NDAT2: FDCAN new data 2 register ------------------------------- */
+/* New data[63:32] */
+#define FDCAN_NDAT2_ND_MASK 0xFFFFFFFF
+#define FDCAN_NDAT2_ND_SHIFT 0
```

Comment:
I don't really see any point to these NDAT1 and NDAT2 defines?  Is this all autogenerated?

Reply:
Sort of. I wrote a helper script to pull bit definitions from a copy/paste from the reference manual. But some manual massaging had to be done.

Happy to pull these definitions out though. I wasn't sure if convention was to define everything even if it had no obvious use just to be consistent or not.